formula1-rest/src/main/webapp/public/components/componentComponentAdd.js

```diff
@@ -0,0 +1,114 @@
+import store from '../../store.js'
+import functions from '../../functions.js'
+
+export default {
+    name: 'Component add',
+
+    props: {
+        id: Number,
+    },
+
+    data() {
+        return {
+            name: null,
+            type: null,
+            errors: {}
+        }
+    },
+
+    setup() {
+        return {store};
+    },
+
+    methods: {
+        validateForm() {
+            let success = true;
+
+            this.errors = [];
+
+            if (!this.name) {
+                this.errors.name = "Name required";
+                functions.showWarningNotification(this.errors.name)
+                success = false;
+            }
+
+            if (!this.type) {
+                this.errors.name = "Type required";
+                functions.showWarningNotification(this.errors.name)
+                success = false;
+            }
+
+            return success;
+        },
+
+        addComponent() {
+            console.log("Add component")
+
+            if (!this.validateForm()) {
+                return;
+            }
+
+            axios.put('http://localhost:8080/pa165/rest/components/', {
+                "name": this.name,
+                "type": this.type,
+            }, {
+                headers: {
+                    "Content-type": "application/json",
+                    "Authorization": `Bearer ${store.$jwt}`,
+                }
+            })
+                .then(response => {
+                    console.log("Success:", response);
+
+                    this.name = response.data.data.name;
```

Comment:
tady je to asi celkove zbytecne ukladat do this.name... hned potom "presmerovavas" na jinou komponentu